azure_functions_worker/utils/dependency.py

```diff
@@ -0,0 +1,313 @@
+        """This is a new implementation of reloading azure and google
+        namespace from customer's .python_packages folder. Only intended to be
+        used in Linux Consumption scenario.
+
+        Parameters
+        ----------
+        cx_working_dir: str
+            The path which contains customer's project file (e.g. wwwroot).
+        """
+        # Specialized working directory needs to be added
+        working_directory: str = os.path.abspath(cx_working_dir)
+
+        # Switch to customer deps and clear out all module cache in worker deps
+        cls._remove_from_sys_path(cls.worker_deps_path)
+        cls._add_to_sys_path(cls.cx_deps_path, True)
+        cls._add_to_sys_path(working_directory, False)
+        logger.info('Reloaded azure google namespace from customer deps')
+
+    @classmethod
+    def _add_to_sys_path(cls, path: str, add_to_first: bool):
+        """This will ensure no duplicated path are added into sys.path and
+        clear importer cache. No action if path already exists in sys.path.
+
+        Parameters
+        ----------
+        path: str
+            The path needs to be added into sys.path.
+            If the path is an empty string, no action will be taken.
+        add_to_first: bool
+            Should the path added to the first entry (highest priority)
+        """
+        if path and path not in sys.path:
+            if add_to_first:
+                sys.path.insert(0, path)
+            else:
+                sys.path.append(path)
+
+            if path in sys.path_importer_cache:
+                sys.path_importer_cache.pop(path)
+            cls._remove_module_cache(path)
+
+    @classmethod
+    def _remove_from_sys_path(cls, path: str):
+        """This will remove path from sys.path and clear importer cache.
+        No action if the path does not exist in sys.path.
+
+        Parameters
+        ----------
+        path: str
+            The path attempts to be removed from sys.path.
+            If the path is an empty string, no action will be taken.
+        """
+        if path and path in sys.path:
+            # Remove all occurances
+            sys.path = list(filter(lambda p: p != path, sys.path))
+            if path in sys.path_importer_cache:
+                sys.path_importer_cache.pop(path)
+            cls._remove_module_cache(path)
+
+    @staticmethod
+    def _get_cx_deps_path() -> str:
+        """Get the directory storing the customer's third-party libraries.
+
+        Returns
+        -------
+        str
+            Core Tools: path to customer's site pacakges
+            Linux Dedicated/Premium: path to customer's site pacakges
+            Linux Consumption: empty string
+        """
+        prefix: Optional[str] = os.getenv(AZURE_WEBJOBS_SCRIPT_ROOT)
+        cx_paths: List[str] = [
+            p for p in sys.path
+            if prefix and p.startswith(prefix) and ('site-packages' in p)
+        ]
```

Comment:
I'm so glad you mention this and I actually thought about it and tested this scenario in custom container.
Since PYTHONPATH will add the path at the first place of sys.path, it will always prioritize loading the packages from PYTHONPATH if it is defined `/python_path_packages/site-packages`. If the packages cannot be found, the worker will try loading the packages from customer's dependencies `/usr/local/lib/python3.7/site-packages`.

![image](https://user-images.githubusercontent.com/48038149/104150086-9168f100-538d-11eb-8131-02456ca44770.png)